tests/app/test_auth.py

```diff
@@ -0,0 +1,117 @@
+"""Tests for the authentication module."""
+
+import base64
+import hmac
+import json
+import time
+from typing import TypedDict
+from unittest.mock import AsyncMock
+
+from fastapi.testclient import TestClient
+
+
+class JWTPayloadDict(TypedDict):
+    """Payload dictionary type."""
+
+    iat: int
+    exp: int
+    jti: str
+
+
+def _get_jwt_token(client_secret: str, payload_dict: JWTPayloadDict) -> str:
+    """Utility to generate a JWT token.
+
+    Args:
+        client_secret (str): Client secret
+        payload_dict (dict[str, str]): Payload dictionary
+
+    Returns:
+        str: a signed JWT token
+    """
+    header_dict = {'typ': 'JWT', 'alg': 'HS256'}
+    header = json.dumps(header_dict)
+    payload = json.dumps(payload_dict)
+
+    header = base64.urlsafe_b64encode(bytes(str(header), 'utf-8')).decode().replace('=', '')
+    payload = base64.urlsafe_b64encode(bytes(str(payload), 'utf-8')).decode().replace('=', '')
+
+    signature = hmac.new(
+        bytes(client_secret, 'utf-8'), bytes(header + '.' + payload, 'utf-8'), digestmod='sha256'
+    ).digest()
+    sigb64 = base64.urlsafe_b64encode(bytes(signature)).decode().replace('=', '')
+
+    token = header + '.' + payload + '.' + sigb64
+    return token
+
+
+def test_credentials_returns_none(client: TestClient, mocker: AsyncMock) -> None:
+    """Test when the credentials call returns None.
+
+    Args:
+        client (TestClient): FastAPI test client
+        mocker: Pytest mocker
+    """
+    mocker.patch('app.auth.HTTPBearer.__call__', return_value=None)
+    response = client.post('/v3/device-registrations')
+    assert response.status_code == 403
+    assert response.json() == {'detail': 'Not authenticated'}
+
+
+def test_missing_authorization_scheme(client: TestClient) -> None:
```

Comment:
I don't understand what this is testing.  Except for the different times, it's identical to the following test.  What is missing?  (This might be deducible if you had a happy path test.)